packages/client/src/workflow-client.ts

```diff
@@ -581,6 +591,69 @@ export class WorkflowClient extends BaseClient {
     }
   }
 
+  /**
+   * Used to add new Build IDs or otherwise update the relative compatibility of Build IDs as
+   * defined on a specific task queue for the Worker Versioning feature. For more on this feature,
+   * see https://docs.temporal.io/workers#worker-versioning
+   *
+   * @param taskQueue The task queue to make changes to.
+   * @param operation The operation to be performed.
+   *
+   * @experimental
+   */
+  public async updateWorkerBuildIdCompatability(taskQueue: string, operation: BuildIdOperation): Promise<void> {
```

Comment:
Make a new TaskQueueClient and put these methods on it and attach it to the top level `Client` instead.

`client.taskQueue.updateBuildIdCompatibility`

Reply:
The word worker will be redundant with a separate client.
Also need a high level API for reachability, put that on the new client.

Reply:
I'm fine adding the reachability client but we didn't in Go/Java since there just wasn't a whole lot to abstract over. It can probably be a little nicer in TS though.